test/Dialog.spec.tsx

```diff
@@ -1,18 +1,14 @@
-import { expect, use as useChaiPlugin } from '@esm-bundle/chai';
-import { cleanup, render } from '@testing-library/react/pure.js';
-import chaiDom from 'chai-dom';
+import { afterEach, describe, expect, it } from 'vitest';
+import { render } from 'vitest-browser-react';
 import sinon from 'sinon';
-import { Dialog, type DialogElement } from '../packages/react-components/src/Dialog.js';
-import createOverlayCloseCatcher from './utils/createOverlayCloseCatcher.js';
+import { Dialog } from '../packages/react-components/src/Dialog.js';
 import { useState } from 'react';
 
-useChaiPlugin(chaiDom);
-
 describe('Dialog', () => {
   const overlayTag = 'vaadin-dialog-overlay';
 
-  const [ref, catcher] = createOverlayCloseCatcher<DialogElement>(overlayTag, (ref) => {
-    ref.opened = false;
+  afterEach(() => {
+    document.querySelector('vaadin-dialog-overlay')?.remove();
```

Comment:
The close catcher construct stopped working only in this test suite for some reason. The whole thing is quite convoluted, I might take another look afterwards if it can be removed entirely and what a good approach would be to ensure overlays are closed after each test.